lib/galaxy/schema/notifications.py

```diff
@@ -0,0 +1,256 @@
+import json
+from datetime import datetime
+from enum import Enum
+from typing import (
+    Any,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    AnyUrl,
+    Field,
+)
+from pydantic.utils import GetterDict
+from typing_extensions import (
+    Annotated,
+    Literal,
+)
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+)
+from galaxy.schema.schema import (
+    CreateTimeField,
+    Model,
+    UpdateTimeField,
+)
+
+
+class NotificationVariant(str, Enum):
+    """The notification variant communicates the intent or relevance of the notification."""
+
+    info = "info"
+    warning = "warning"
+    urgent = "urgent"
+
+
+# Register a new category by adding it to the corresponding Enum:
+# - NotificationCategory: these notification categories will be always received by the user.
+# - OptionalNotificationCategory: these notifications can be opt-out by the user.
+# Then register the content model further down below.
+
+
+class MandatoryNotificationCategory(str, Enum):
+    """These notification categories cannot be opt-out by the user.
+
+    The user will always receive notifications from these categories.
+    """
+
+    broadcast = "broadcast"
+
+
+class PersonalNotificationCategory(str, Enum):
+    """These notification categories can be opt-out by the user and will be
+    displayed in the notification preferences.
+    """
+
+    message = "message"
+    near_quota_limit = "near_quota_limit"
+    new_history_shared = "new_history_shared"
+    new_workflow_shared = "new_workflow_shared"
+    new_page_shared = "new_page_shared"
+    new_visualization_shared = "new_page_shared"
+    workflow_execution_completed = "workflow_execution_completed"
+
+
+NotificationCategory = Union[MandatoryNotificationCategory, PersonalNotificationCategory]
+
+
+class NotificationContentBase(Model):
+    subject: str
+    message: str  # markdown?
+
+
+class ActionLink(Model):
+    action_name: str
+    link: AnyUrl
```

Comment:
Can we not do urls, but return entity ids, so the client can build whatever operation is appropriate for an entity ?

So if it's a workflow_id, you could view, edit, run, download etc ... ?

Take a look at `WarningReason`, `FailureReason`, `CancelReason` ? That way we also don't need to hook into the backend routing for generating routes

Reply:
Ohh wait, I see what you mean, the `share_link` below should go :sweat_smile: